app/models/container_project.rb

```diff
@@ -10,22 +10,22 @@ class ContainerProject < ApplicationRecord
   include Purging
   belongs_to :ext_management_system, :class_name => "ManageIQ::Providers::ContainerManager", :foreign_key => "ems_id", :inverse_of => :container_projects
   has_many :container_groups, -> { active }, :inverse_of => :container_project
-  has_many :container_routes
-  has_many :container_replicators
-  has_many :container_services
+  has_many :container_routes, :dependent => :destroy
+  has_many :container_replicators, :dependent => :destroy
+  has_many :container_services, :dependent => :destroy
   has_many :containers, :through => :container_groups
   has_many :container_images, -> { distinct }, :through => :container_groups
   has_many :container_nodes, -> { distinct }, :through => :container_groups
   has_many :container_quotas, -> { active }, :inverse_of => :container_project
   has_many :container_quota_scopes, :through => :container_quotas
   has_many :container_quota_items, :through => :container_quotas
-  has_many :container_limits
+  has_many :container_limits, :dependent => :destroy
   has_many :container_limit_items, :through => :container_limits
-  has_many :container_builds
-  has_many :container_templates
+  has_many :container_builds, :dependent => :destroy
+  has_many :container_templates, :dependent => :destroy
   has_many :all_container_groups, :class_name => "ContainerGroup", :inverse_of => :container_project
   has_many :archived_container_groups, -> { archived }, :class_name => "ContainerGroup"
-  has_many :persistent_volume_claims
+  has_many :persistent_volume_claims, :dependent => :destroy
```

Comment:
WAT, yes, we probably don't have that many container projects but still, I think these should all be destroyed... otherwise the purger for projects will just leave these orphaned.  

@agrare do any of these have a chance to be impossible to delete in the UI/backend due to many tens of thousands of rows?  Maybe builds?  I'm not sure.

Reply:
`persistent_volume_claims` are associated to a `container_volume` which does the dependent destroy,
```
app/models/container_volume.rb:  belongs_to :persistent_volume_claim, :dependent => :destroy
```